extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -1574,6 +1619,40 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     dev().warn(TAG, message, error);
   }
 
+  /**
+   * Generate a 32-byte random string.
+   * Uses the win.crypto when available.
+   * @return {string} The random string
+   * @private
+   */
+  getRandomString_() {
+    // 16 hex characters * 2 bytes per character = 32 bytes
+    const length = 16;
+    let randomValues = Services.cryptoFor(this.win).getSecureRandomBytes(
+      length
+    );
+
+    if (!randomValues) {
```

Comment:
You can combine these into a single iteration which IMO is more readable:

let randomSubdomain = '';
for (let i = 0; i < length; i++) {
   // If crypto isn't available, just use Math.random.
   const randomValue = randomValues ? randomValues[i] || Math.floor(Math.random() * 255);
   // Ensure each byte is represented with two hexadecimal characters.
    if (randomValue <= 15) {
        randomSubdomain += '0';
    }
    randomSubdomain += randomValue.toString(16);
}
return randomSubdomain;

Reply:
done